napi-inl.h

```diff
@@ -2595,12 +2595,110 @@ inline Error::Error(napi_env env, napi_value value) : ObjectReference(env, nullp
   if (value != nullptr) {
     napi_status status = napi_create_reference(env, value, 1, &_ref);
 
+    // Creates a wrapper object containg the error value (primitive types) and
+    // create a reference to this wrapper
+    if (status != napi_ok) {
+      napi_value wrappedErrorObj;
+      status = napi_create_object(env, &wrappedErrorObj);
+
+      NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_create_object");
+
+      napi_property_descriptor errValDesc = {
+          "errorVal",                     // const char* utf8Name
+          String::From(env, "errorVal"),  //  napi_value name
+          nullptr,                        // Method
+          nullptr,                        // getter
+          nullptr,                        // setter
+          Value::From(env, value),        // napi_value value
+          napi_enumerable,
+          nullptr};
+
+      status = napi_define_properties(env, wrappedErrorObj, 1, &errValDesc);
+      NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_define_properties");
+
+      MaybeOrValue<Symbol> result = Symbol::For(env, "isWrapObject");
+      napi_property_descriptor wrapObjFlag = {nullptr,
+                                              nullptr,
+                                              nullptr,
+                                              nullptr,
+                                              nullptr,
+                                              Value::From(env, value),
+                                              napi_enumerable,
+                                              nullptr};
+
+#ifdef NODE_ADDON_API_ENABLE_MAYBE
+      Symbol uniqueSymb;
+      if (result.IsJust()) {
+        uniqueSymb = result.Unwrap();
+      }
+
+      wrapObjFlag.name = uniqueSymb;
+      status = napi_define_properties(env, wrappedErrorObj, 1, &wrapObjFlag);
+      NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_define_properties");
+#else
+
+      wrapObjFlag.name = result;
+      status = napi_define_properties(env, wrappedErrorObj, 1, &wrapObjFlag);
+      NAPI_FATAL_IF_FAILED(status, "Error::Error", "napi_define_properties");
```

Comment:
3. Move these two statements below the `#endif`. That way, you are not duplicating them in both bodies of the `#if...#else...#endif`, thereby reducing the size of the the `#if...#else...#endif` block.